src/core/action-creator.ts

```diff
@@ -51,7 +51,21 @@ export function createActions(flux: FluxCapacitor) {
                 func: ({ query }) => !('query' in search) || !!query || query === null,
                 msg: 'search term is empty'
               }, {
-                func: ({ query }, state) => query !== Selectors.query(state) || query === null,
+                  // currently assume that search will only ever have up to one refinement
+                  func: ({ query, value, navigationId }, state) => {
+                  let currentRefinements = Selectors.selectedRefinements(state);
+                  const searchHasRefinement = Boolean(value && navigationId);
+                  // query is not the same
+                  return query !== Selectors.query(state) ||
+                  // xor for either there exists refinements or the current search has refinement
+                  currentRefinements.length > 0 !== searchHasRefinement ||
+                  // current search has refinement and does not match any of the previous refinements
+                  searchHasRefinement &&
+                    // try to find any refinement not matching the refinement in the query
+                    Selectors.selectedRefinements(state).find(
```

Comment:
do we need to re-select `Selector.selectedRefinements(state)` when we have `currentRefinements`?